jax/numpy/lax_numpy.py

```diff
@@ -524,6 +524,21 @@ def _float_divmod(x1, x2):
 
 @_wraps(onp.power)
 def power(x1, x2):
+  # Special case for small positive integer scalars: use binary exponentiation.
+  # Using lax.pow may be imprecise for floating-point values; the goal of this
+  # code path is to make sure we end up with a precise output for the common
+  # pattern ``x ** 2`` or similar.
+  if isinstance(x2, int) and x2 >= 0 and x2 <= 64:
+    x1 = asarray(x1)
+    acc = None
+    while x2 > 0:
+      if x2 & 1:
+        acc = x1 if acc is None else lax.mul(acc, x1)
+      x2 >>= 1
+      if x2 > 0:
+        x1 = lax.mul(x1, x1)
```

Comment:
I would be reassured if you added a test for all integer powers in some range, e.g., 0 to 8 inclusive.

Reply:
I already did ...?

Reply:
yep I was looking at the older commit